admin/angel/Core.c

```diff
@@ -189,14 +200,33 @@ static void initTunfd(Dict* args, void* vcontext, String* txid, struct Allocator
         struct Pipe* p = Pipe_forFiles(fileno, fileno, ctx->base, &jmp.handler, ctx->alloc);
         p->logger = ctx->logger;
         if (type == FileNo_Type_ANDROID) {
+            Log_info(ctx->logger, "Plumbing AndroidWrapper iface in initTunFd() request [%d] [%d]",
+                     (int)*tunfd, (int)*tuntype);
             struct AndroidWrapper* aw = AndroidWrapper_new(ctx->alloc, ctx->logger);
             Iface_plumb(&aw->externalIf, &p->iface);
             Iface_plumb(&aw->internalIf, &ctx->nc->tunAdapt->tunIf);
         } else {
+            Log_info(ctx->logger, "Plumbing iface in initTunFd() request [%d] [%d]",
+                     (int)*tunfd, (int)*tuntype);
             Iface_plumb(&p->iface, &ctx->nc->tunAdapt->tunIf);
         }
+        Log_info(ctx->logger, "Finished initTunFd() request [%d] [%d]", (int)*tunfd, (int)*tuntype);
+
+        String* path = String_printf(ctx->alloc, "/proc/%d/fd", getpid());
+        DIR* dir = opendir(path->bytes);
+        if (!dir) {
+            Log_warn(ctx->logger, "Unable to opendir() [%s] [%s]", path->bytes, strerror(errno));
+        } else {
+            struct dirent* ent;
+            while ((ent = readdir(dir))) {
+                Log_debug(ctx->logger, "[type=%d] %s/%s", ent->d_type, path->bytes, ent->d_name);
+            }
+            closedir(dir);
+        }
+
```

Comment:
Yea it's fair that all this don't belong in Core. Do you mean to replace `"/proc/%d/fd"`, `opendir()`, `readdir()`, and `closedir()` with **uv.h** interfaces, and move this part to **util/events/libuv**? Core will just import **util/events/FileSystem.h** then.

I guess the existing `#include "util/events/libuv/FileNo_admin.h"` in **Core.c** should also be **util/events/FileNo_admin.h**?